sdk/python/lib/pulumi/runtime/resource.py

```diff
@@ -132,7 +132,7 @@ class ResourceResolverOperations(NamedTuple):
     log.debug(f"resource {props} prepared")
     return ResourceResolverOperations(
         parent_urn,
-        serialized_props,
+        cast(Dict[str, Any], serialized_props),
```

Comment:
Is this true that a `Struct` is a `Dict[str, Any]`?  Feels like there may actually be some deeper bug here?

Reply:
Our tests fail if I call `MessageToDict` so it isn't true. It looks like changing the type of this property in ResourceResolverOptions to `pb2_struct.Struct` works, so I'll make that change.